pkgs/data/fonts/nerdfonts/default.nix

```diff
@@ -25,33 +26,97 @@ let
+    "LiberationMono" = "liberation-mono";
+    "Meslo" = "meslo";
+    "Monofur" = "monofur";
+    "Monoid" = "monoid";
+    "Mononoki" = "mononoki";
+    "MPlus" = "mplus";
+    "Noto" = "noto";
+    "OpenDyslexic" = "open-dyslexic";
+    "Overpass" = "overpass";
+    "ProFont" = "profont";
+    "ProggyClean" = "proggy-clean";
+    "RobotoMono" = "roboto-mono";
+    "ShareTechMono" = "share-tech-mono";
+    "SourceCodePro" = "source-code-pro";
+    "SpaceMono" = "space-mono";
+    "Terminus" = "terminus";
+    "Tinos" = "tinos";
+    "Ubuntu" = "ubuntu";
+    "UbuntuMono" = "ubuntu-mono";
+    "VictorMono" = "victor-mono";
+  };
+  # Create a separate derivation for each individual font so that a font doesn't
+  # need to be rebuilt when used in different selection fonts. Reduces
+  # unnecessary rebuilds and disk usage.
+  fontPkgs = lib.attrsets.mapAttrs' (
     fName:
     fSha:
-    (fetchurl {
-      url = "https://github.com/ryanoasis/nerd-fonts/releases/download/v${version}/${fName}.zip";
-      sha256 = fSha;
+    lib.nameValuePair (attrNames."${fName}") (stdenv.mkDerivation rec {
+      inherit version;
+      pname = "nerdfonts-${fName}";
+      src = fetchurl {
+        url = "https://github.com/ryanoasis/nerd-fonts/releases/download/v${version}/${fName}.zip";
+        sha256 = fSha;
+      };
+      nativeBuildInputs = [
+        unzip
+      ];
+      sourceRoot = ".";
+      installPhase = ''
+        find -name \*.otf -exec mkdir -p $out/share/fonts/opentype/NerdFonts \; -exec mv {} $out/share/fonts/opentype/NerdFonts \;
+        find -name \*.ttf -exec mkdir -p $out/share/fonts/truetype/NerdFonts \; -exec mv {} $out/share/fonts/truetype/NerdFonts \;
+      '';
+      meta = with stdenv.lib; {
+        description = "${fName} font in Nerdfonts collection";
+        homepage = "https://nerdfonts.com/";
+        maintainers = with maintainers; [ doronbehar ];
+        hydraPlatforms = []; # 'Output limit exceeded' on Hydra
+      };
     })
   ) selectedFontsShas;
-in
-
-stdenv.mkDerivation rec {
-  inherit version;
-  inherit srcs;
-  pname = "nerdfonts";
-  nativeBuildInputs = [
-    unzip
-  ];
-  sourceRoot = ".";
-  buildPhase = ''
-    echo "selected fonts are ${toString selectedFonts}"
-    ls *.otf *.ttf
-  '';
-  installPhase = ''
-    find -name \*.otf -exec mkdir -p $out/share/fonts/opentype/NerdFonts \; -exec mv {} $out/share/fonts/opentype/NerdFonts \;
-    find -name \*.ttf -exec mkdir -p $out/share/fonts/truetype/NerdFonts \; -exec mv {} $out/share/fonts/truetype/NerdFonts \;
-  '';
-
+in symlinkJoin {
+  # This package is just a thin symlink join which wraps the separate font
+  # packages under the same store path.
+  name = "nerdfonts-${version}";
+  paths = lib.attrValues fontPkgs;
+  # Make each font available also as attributes of nerdfonts package. That is,
+  # individual font packages can be accessed as nerdfonts._3270,
+  # nerdfonts.agave, nerdfonts.anonymous-pro and so on.
+  passthru = fontPkgs;
```

Comment:
Interesting.... This doesn't really work if you try e.g:

```
nix-build -A nerdfonts.agave
```

And you haven't built `nerdfonts` with "Agave" ever..

I'm afraid this also explains the issue of these fonts not found in `nix search -u nerdfonts`.